include/aws/common/json.h

```diff
@@ -166,6 +166,31 @@ bool aws_json_value_has_key(const struct aws_json_value *object, struct aws_byte
  */
 AWS_COMMON_API
 int aws_json_value_remove_from_object(struct aws_json_value *object, struct aws_byte_cursor key);
+
+/**
+ * @brief callback for iterating keys of an object
+ * return false to exit early
+ */
+typedef bool(aws_json_on_key_encountered_const_fn)(
+    const struct aws_byte_cursor *key,
+    const struct aws_json_value *value,
+    void *user_data);
+
+/**
+ * @brief iterates though fields of the object.
+ * iteration is sequential in order fields were initially parsed.
+ * @param object object to iterate over.
+ * @param on_key callback for when key is endcountered.
+ * @param user_data user data to pass back in callback.
+ * @return AWS_OP_SUCCESS when iteration finishes completely or exits early,
+ * AWS_OP_ERR if value is not an object.
+ */
+AWS_COMMON_API
+int aws_json_const_iterate_object(
+    struct aws_json_value *object,
+    aws_json_on_key_encountered_const_fn *on_key,
```

Comment:
"on_member" probably a better name than "on_key" for the typedef and param.

according to json.org, and "object" is composed of "members"